backend/webdav/webdav.go

```diff
@@ -1199,9 +1241,114 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
 		_ = o.Remove(ctx)
 		return err
 	}
-	// read metadata from remote
-	o.hasMetaData = false
-	return o.readMetaData(ctx)
+	return nil
+
+}
+
+// Chunked update for Nextcloud (see https://docs.nextcloud.com/server/20/developer_manual/client_apis/WebDAV/chunking.html)
+func (o *Object) updateChunked(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) (err error) {
+
+	var uploadDir string = strings.ReplaceAll(path.Join(o.fs.root, o.remote), "/", "_")
+
+	hasher := md5.New()
+	_, err = hasher.Write([]byte(uploadDir))
+
+	if err != nil {
+		return errors.Wrap(err, "Chunked upload couldn't hash URL")
+	}
+
+	uploadDir = "chunked-upload-" + hex.EncodeToString(hasher.Sum(nil))
+
+	saveRoot := o.fs.root
+	saveRemote := o.remote
+	o.fs.root = "/"
+	o.fs.srv.SetRoot(strings.Replace(o.fs.endpointURL, "files", "uploads", 1))
+
+	err = o.fs.mkdir(ctx, uploadDir)
+	if err != nil {
+		defer atexit.OnError(&err, func() {
+			// Try to abort the upload, but ignore the error.
+			fs.Debugf(src, "Cancelling chunked upload")
+			_ = o.fs.pacer.Call(func() (bool, error) {
+				err := o.fs.Rmdir(ctx, uploadDir)
+				return o.fs.shouldRetry(nil, err)
+			})
+		})()
+		return errors.Wrap(err, "Update mkUploadDir failed")
```

Comment:
Maybe `errors.Wrap(err, "making upload directory failed")`

Note that comments in `errorrs.Wrap` should start with a lowercase letter.